### PR TITLE
Fix: Momentum tracker overlaps A/V dock in some configurations

### PR DESCRIPTION
**Pull request by WesWedding**
Tentative fix for #104 , needs testing in v11 and v10 to be sure.

### CODE REVIEW

File: src/module/sta.js [hunk 1]
```diff
@@ -318,15 +319,16 @@ Hooks.once('init', function() {
   });
 
   Hooks.on('ready', function() {
-    const t = new STATracker();
-    renderTemplate('systems/sta/templates/apps/tracker.hbs').then((html) => {
```
Comment (WesWedding) [inline src/module/sta.js:322]: This call to renderTemplate() seemed to serve no purpose, so I removed it.

**Comment by WesWedding** [comment]
Testing in v11: Works as expected.
Testing in v10: Works as expected.

New use-case discovered that needs to be accounted for: You can pop out the camera from the CameraViews element which completely collapses the CameraViews element.  Current behavior doesn't catch this use case, so the tracker is left floating way out of place.

**Comment by tokeidlom** [comment]
Ok, so given this a test, first thing is, it works! The only thing of note is that it reacts to the camera dock position setting regardless of whether the camera is on, which caught me out when I first loaded it as I had it set the camera to left prevously and found the tracker floating in the middle of the screen. Otherwise, any reason for the dock to move right instead of down when the dock is bottom?

It has a minor issue with the LCARS module, but I suspect that can be easily fixed by adding some extra css to that module:
image

**Comment by WesWedding** [comment]
I think I need more clear reproduction steps, I'm not following.

**Comment by WesWedding** [comment]
I think I need to write up some test cases for this one.

**Comment by tokeidlom** [comment]
Steps to reproduce are:
Click on settings, Audio/Video configuration, with Audio/Video conferencing mode disabled change the dock position to right, the momentum tracker moves to the left despite there being no AV bar on screen, same behaviour for bottom.

This may be as simple to fix as adding an extra check to the existing PR if ((av dock right) && (AV enabled)) {move the tracker left}

**Comment by tokeidlom** [comment]
Ok, looks good with 2 caveats / queries:
1) When the AV bar is at the bottom, the tracker now minimises right intead of down, any reason for that?
2) The tracker is by default now set a little lower on the screen, almost in line with the bottom of the screen, I agree that generally looks better, but, I think it was set to be in line with the top edge of the macro bar previously for people with lower resolution screens, so that they don't conflict.

**Comment by WesWedding** [comment]
The 2nd bit is unintentional, I'll have to compare it to the unmodified version again.

Checking the 1st.

**Comment by WesWedding** [comment]
Oh, I misunderstood your first question.

The reason that the hide function changes direction in that case is because I felt like there was visual and usability problems keeping it as-is.


image

I think that just looks kinda weird, with it just kind of lurking behind a translucent element like that.  Also kinda just floating there detached from anything else.

I also wanted to avoid users trying to click the still-visible tracker in an attempt to use it.

**Comment by WesWedding** [comment]
Re point 1: I cannot reproduce. 

image 
image

Develop branch (1st image) vs PR branch (2nd image)

**Comment by WesWedding** [comment]
When the dock was on the bottom the tracker was too low, though, fixed that issue.

**Comment by tokeidlom** [comment]
You are quite right, the LCARS UI makes the box go above the macro bar, not the default UI. Which is perhaps something we could fix in the future too as I tried it on a small screen and they do cover each other.
The moving right does look better in this case (but causes the problem mentioned below).
This update does break LCARS UI somewhat, it does make it fly off into the middle of the screen disconnected from anything else when the AV bar is right. Also it has the collapse button on the right, so with the AV bottom, once pressed it disappears behind the chat bar and cannot be clicked again. Neither of these things are your issue to fix of course, just mentioning them.

**Comment by tokeidlom** [comment]
image
Image for illustrative purposes, this is an existing issue though not really related to this fix, although with the AV bar out it makes it more likely to occur.

**Review by tokeidlom — APPROVED** [review]
Tested, and seems to work as intended, need to work out how to apply this to LCARS, and happy to receive any support in that if it is forthcoming!

**Comment by tokeidlom** [comment]
I've fixed LCARS module to work with this properly, so need need for help. Thank you.

**Comment by WesWedding** [comment]
Glad it wasn't too much trouble to get it to fit into LCARS!

The overlap with the macro bar is definitely a potential issue for some users;  I'm torn between at least acknowledging the problem in an Issue but not fixing it, or just ignoring the issue until someone has a problem.

**Comment by tokeidlom** [comment]
Its been an issue since forever, and I am just mentioning it now, it also doesn't happen with LCARS UI (not sure the split of those who use LCARS vs traditional UI). So, I would suggest we go for the second option and see if anyone raises it in the future. Any change would be a compromise anyway as someone else will complain it is now too high and obstructing their view of venus or something!

**Pull request merged**